src/librustc_resolve/lib.rs

```diff
@@ -145,7 +147,7 @@ pub enum ResolutionError<'a> {
     /// error E0411: use of `Self` outside of an impl or trait
     SelfUsedOutsideImplOrTrait,
     /// error E0412: use of undeclared
-    UseOfUndeclared(&'a str, &'a str),
+    UseOfUndeclared(&'a str, &'a str, Vec<Path>),
```

Comment:
Alternatively, make a new type `struct Suggestions(Vec<Path>)` or something so it is self-explanatory